### PR TITLE
Implement stair-case Yee solver with EB in RZ geometry

### PR DESCRIPTION
**Pull request by RemiLehe**
This implements the Yee solver with EB. The code is similar to the corresponding Cartesian code: in the cells that are covered with EB cells, the E field is simply not updated.

I added a test where a laser is diffracted by a cylindrical object (represented by the EB). The laser is emitted from the bottom, in the plots below:
Figure 80
The test then checks that the position of the first dark ring (i.e. first minimum) corresponds to the theoretical prediction (Airy pattern).

### CODE REVIEW

**Comment by lgiacome** [comment]
This is great Remi! I've had a look at the changes and they look good to me. I've seen that you've enabled also the ECT solver. Should we add a test for that too? For example, we could simply increase the height of the cylinder by e.g. dz/5, which would give some "unstable faces". I don't know if RZ+ECT is really interesting but maybe it could be of help in the future to simulate conic geometries. We can also add this test in a follow-up PR.

**Comment by lgiacome** [comment]
Sorry I got confused. It seems like the ECT initialization can be done also in RZ, but the solver doesn't support it yet. Can you confirm?

**Comment by RemiLehe** [comment]
@lgiacome Thanks for the review. 
I agree the PR is confusing, sorry. I enabled some of the EB structures in RZ without realizing that they were specific to the ECT algorithm. You are correct that the ECT algorithm is not enabled in RZ, including in this PR. I will revert some of these changes in order to make this less confusing.

**Comment by lgiacome** [comment]
Thanks a lot for the clarification Remi, I've had another look and everything looks great to me.
Should we try to run the CI?

File: Source/FieldSolver/FiniteDifferenceSolver/EvolveE.cpp
Comment (roelof-groenewald) [inline Source/FieldSolver/FiniteDifferenceSolver/EvolveE.cpp]: In the Ohm's law solver I used `lr` rather than `lx` to make it clear that this is actually an edge length in r. It could be nice to have the same convention between the two solvers but of course it is also not really needed.

**Review by roelof-groenewald — APPROVED** [review]
Looks great! It's awesome to have a support for even more use cases now!

File: Source/FieldSolver/FiniteDifferenceSolver/EvolveE.cpp
Comment (roelof-groenewald) [reply Source/FieldSolver/FiniteDifferenceSolver/EvolveE.cpp]: I won't merge yet and leave it up to you if you want to rename `lx` -> `lr` 😄

Comment (RemiLehe) [reply Source/FieldSolver/FiniteDifferenceSolver/EvolveE.cpp]: Good point. I will modify the PR accordingly.

**Pull request merged**
